### PR TITLE
update permissionedExchange contract and tests

### PR DESCRIPTION
**Pull request by Jianzhuo**
refer Support update kSQT order pool in permissionExchange contract #73
consider use tokenGiveBalance instead of amountGiveBalance

### CODE REVIEW

File: contracts/PermissionedExchange.sol [hunk 1]
```diff
@@ -89,24 +91,42 @@ contract PermissionedExchange is Initializable, OwnableUpgradeable {
         address _tokenGet,
         uint256 _amountGive,
         uint256 _amountGet,
-        uint256 _expireDate
+        uint256 _expireDate,
+        uint256 _pairId,
+        uint256 _balance
```
Comment (mzxyz) [inline contracts/PermissionedExchange.sol]: To make it clear, change `_balance` to `_tokenGiveBalance`

File: contracts/PermissionedExchange.sol [hunk 2]
```diff
@@ -89,24 +91,42 @@ contract PermissionedExchange is Initializable, OwnableUpgradeable {
...
     ) public onlyOwner {
         require(_expireDate > block.timestamp, 'invalid expireDate');
         require(_amountGive > 0 && _amountGet > 0, 'invalid amount');
-        IERC20(_tokenGive).safeTransferFrom(msg.sender, address(this), _amountGive);
+        if(_balance > 0){
```
Comment (mzxyz) [inline contracts/PermissionedExchange.sol]:
```suggestion
        if (_balance > 0){
```

File: contracts/PermissionedExchange.sol [hunk 3]
```diff
@@ -89,24 +91,42 @@ contract PermissionedExchange is Initializable, OwnableUpgradeable {
...
+    function createPairOrders(
+        address _tokenGive,
+        address _tokenGet,
+        uint256 _amountGive,
+        uint256 _amountGet,
+        uint256 _expireDate,
+        uint256 _balance
+    ) public onlyOwner {
+        require(_balance > 0, 'pair orders should have balance');
+        sendOrder(_tokenGive, _tokenGet, _amountGive, _amountGet, _expireDate, nextOrderId+1, _balance);
```
Comment (HZHAIX) [inline contracts/PermissionedExchange.sol:126]:
```suggestion
        sendOrder(_tokenGive, _tokenGet, _amountGive, _amountGet, _expireDate, nextOrderId, _balance);
```

File: contracts/PermissionedExchange.sol [hunk 4]
```diff
@@ -89,24 +91,42 @@ contract PermissionedExchange is Initializable, OwnableUpgradeable {
...
+        sendOrder(_tokenGet, _tokenGive, _amountGet, _amountGive, _expireDate, nextOrderId-1, 0);
```
Comment (HZHAIX) [inline contracts/PermissionedExchange.sol:127]:
```suggestion
        sendOrder(_tokenGet, _tokenGive, _amountGet, _amountGive, _expireDate, nextOrderId+1, 0);
```

File: contracts/PermissionedExchange.sol [hunk 5]
```diff
@@ -89,24 +91,42 @@ contract PermissionedExchange is Initializable, OwnableUpgradeable {
...
         nextOrderId += 1;
     }
 
+    function createPairOrders(
+        address _tokenGive,
+        address _tokenGet,
+        uint256 _amountGive,
+        uint256 _amountGet,
+        uint256 _expireDate,
+        uint256 _balance
```
Comment (mzxyz) [inline contracts/PermissionedExchange.sol]: To make it clear, change _balance to _tokenGiveBalance

**Pull request merged**